conans/client/cmd/build.py

```diff
@@ -44,6 +44,8 @@ def cmd_build(app, conanfile_path, source_folder, build_folder, package_folder,
         conan_file.layout.set_base_build_folder(build_folder)
         conan_file.layout.set_base_source_folder(source_folder)
         conan_file.layout.set_base_package_folder(package_folder)
+        conan_file.layout.set_base_generators_folder(base_path)
```

Comment:
if `layout.generator.folder` is adjusted, that is relative to the base. Otherwise it uses the `install_folder`.
Note: All these folder parameters (source_folder, build_folder, package_folder...) should disappear in Conan 2.0